lessons/02-javascript/README.md

```diff
@@ -152,6 +157,8 @@ for (var key in object) {
 }
 ```
 
+Note that the order of iteration is not guarenteed with this syntax and later, if you use libraries which modify the prototype (more on this later) of base types, can have strange behavior.
```

Comment:
Yup, "native" ("barebones"?) is obviously the fastest choice, but regardless I can't find a good reason to use for...in with hasOwnProperty over Object.keys